### PR TITLE
Functions and scope questions

### PR DESCRIPTION
**Pull request by KarnPratapSingh**

### CODE REVIEW

File: function_and_scope_questions.js [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+seven(times(five())); // must return 35
+four(plus(nine())); // must return 13
+eight(minus(three())); // must return 5
+six(dividedBy(two())); // must return 3 */
+
+// Solution one:
+
+function zero(operation) {
+    if (operation) {
+        return parseInt(eval(`0${operation}`));
```
Comment (suraj-pareek) [inline function_and_scope_questions.js]: eval () is costly, do we really need to use it?

File: function_and_scope_questions.js [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+ If the word's length is even, return the middle 2 characters. */
+
+
+
+//SOLUTION:
+
+function getMiddlePartofInputString(inputString) {
+    //Code goes here!
+    if (inputString.length % 2 == 0) {
+        return inputString.substring(((inputString.length / 2) - 1), ((inputString.length / 2) + 1));
```
Comment (suraj-pareek) [inline function_and_scope_questions.js]: please divide conditions in a separate function;

File: function_and_scope_questions.js [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+// at the end, returning the index of the first true value
+
+function partitionOn(items) {
+
+
+    var newarray = [];
+    function firstFillAllOddNumbers(items) {
+        items.forEach(item => {
+            if (item % 2 != 0) {
+                newarray.push(item);
```
Comment (suraj-pareek) [inline function_and_scope_questions.js]: please use strictly type checking here;

File: function_and_scope_questions.js [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+
+    var booleanArray = [];
+    newarray.forEach(element => {
+        booleanArray.push(element % 2 == 0);
+    }); // Filling boolean_Array with Boolean Values
+
+
+    var indexOfFirstTrueValue = booleanArray.indexOf(true);
+
+    for (var i = 0; i < items.length; i++) {
```
Comment (suraj-pareek) [inline function_and_scope_questions.js]: please avoid using variables names like i/j;

**Review by suraj-pareek — CHANGES_REQUESTED** [review]

**Review by suraj-pareek — CHANGES_REQUESTED** [review]
Please address the last review comments;

File: function_and_scope_questions.js [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+  }
+  function dividedBy(x) {
+    return function (y) {
+      return Math.floor(y / x);
+    };
+  }
+  
+
+  function runSolutionOne(){
+    let answer=seven(times(five()));
```
Comment (suraj-pareek) [inline function_and_scope_questions.js]: please beautify this file

File: function_and_scope_questions.js [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+
+
+
+//SOLUTION:
+
+function getMiddlePartofInputString(inputString) {
+    //Code goes here!
+    
+    let halfLengthOfString=inputString.length / 2;
+    let lengthIsEven=inputString.length % 2 == 0;
```
Comment (suraj-pareek) [inline function_and_scope_questions.js]: please use strict equal

File: function_and_scope_questions.js [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+        items.forEach(item => {
+            if (item % 2 !== 0) {
+                newarray.push(item);
+            }
+        });
+    }
+
+    function secondFillAllEvenNumbers(items) {
+        items.forEach(item => {
+            if (item % 2 == 0) {
```
Comment (suraj-pareek) [inline function_and_scope_questions.js]: please use strict equal

File: function_and_scope_questions.js [hunk 8]
```diff
@@ -0,0 +1,199 @@
...
+    }
+    firstFillAllOddNumbers(items);
+    secondFillAllEvenNumbers(items);
+
+
+
+
+    var booleanArray = [];
+    newarray.forEach(element => {
+        booleanArray.push(element % 2 == 0);
```
Comment (suraj-pareek) [inline function_and_scope_questions.js]: please use strict equal

File: function_and_scope_questions.js [hunk 9]
```diff
@@ -0,0 +1,199 @@
...
+    }); // Filling boolean_Array with Boolean Values
+
+
+    var indexOfFirstTrueValue = booleanArray.indexOf(true);
```
Comment (suraj-pareek) [inline function_and_scope_questions.js]: can we have some other name? This doesn't sound good;

**Review by suraj-pareek — CHANGES_REQUESTED** [review]

File: function_and_scope_questions.js (hunk 9 above)
Comment (KarnPratapSingh) [reply function_and_scope_questions.js]: One of the two purposes of the program was to give the index of first true vale in the array, so only writing "index" or "firstTrueIndex" isn't fitting correctly. 
Which is why I had no other option but to write the complete "indexOfFirstTrueValue"

**Review by suraj-pareek — APPROVED** [review]
Good Job!

**Pull request closed without merging**